ios/VisionCameraImageLabeler.m

```diff
@@ -1,51 +1,24 @@
-//
-//  VisionCameraImageLabeler.m
-//  VisionCameraExample
-//
-//  Created by Marc Rousavy on 06.05.21.
-//
-
+#import <Foundation/Foundation.h>
 #import <VisionCamera/FrameProcessorPlugin.h>
-#import <VisionCamera/Frame.h>
-#import <MLKit.h>
-
-// Example for an Objective-C Frame Processor plugin
+#import <VisionCamera/FrameProcessorPluginRegistry.h>
 
-@interface QRCodeFrameProcessorPluginObjC : NSObject
+#if defined __has_include && __has_include("VisionCameraImageLabeler-Swift.h")
+#import "VisionCameraImageLabeler-Swift.h"
+#else
+#import "VisionCameraImageLabeler/VisionCameraImageLabeler-Swift.h"
+#endif
 
-+ (MLKImageLabeler*) labeler;
+// This macro didn't properly register the frame processor plugin in the registry
+// VISION_EXPORT_SWIFT_FRAME_PROCESSOR(VisionCameraImageLabeler, imageLabeler)
 
+@interface VisionCameraImageLabeler (FrameProcessorPluginLoader)
 @end
 
-@implementation QRCodeFrameProcessorPluginObjC
-
-+ (MLKImageLabeler*) labeler {
-  static MLKImageLabeler* labeler = nil;
-  if (labeler == nil) {
-    MLKImageLabelerOptions* options = [[MLKImageLabelerOptions alloc] init];
-    labeler = [MLKImageLabeler imageLabelerWithOptions:options];
-  }
-  return labeler;
-}
-
-static inline id labelImage(Frame* frame, NSArray* arguments) {
-  MLKVisionImage *image = [[MLKVisionImage alloc] initWithBuffer:frame.buffer];
-  image.orientation = frame.orientation; // <-- TODO: is mirrored?
-
-  NSError* error;
-  NSArray<MLKImageLabel*>* labels = [[QRCodeFrameProcessorPluginObjC labeler] resultsInImage:image error:&error];
-
-  NSMutableArray* results = [NSMutableArray arrayWithCapacity:labels.count];
-  for (MLKImageLabel* label in labels) {
-    [results addObject:@{
-      @"label": label.text,
-      @"confidence": [NSNumber numberWithFloat:label.confidence]
-    }];
-  }
-
-  return results;
+@implementation VisionCameraImageLabeler (FrameProcessorPluginLoader)
++ (void) load {
+  [FrameProcessorPluginRegistry addFrameProcessorPlugin:@"imageLabeler"
+    withInitializer:^FrameProcessorPlugin*(VisionCameraProxyHolder* proxy, NSDictionary* options) {
+    return [[VisionCameraImageLabeler alloc] initWithProxy:proxy withOptions:options];
+  }];
```

Comment:
I was unable to use the macro here that was defined in the [documentation[(https://react-native-vision-camera.com/docs/guides/frame-processors-plugins-ios#manual-setup) (under the manual, Swift tab).

This is where I hit the snag that `imageLabeler` plugin wasn't registered.

The macro doesn't quite generate the version that is working here. Unsure if the macro is incorrect, but you likely have more context around that.

Happy to fix it if it is an issue - I found a related issue over here: https://github.com/rodgomesc/vision-camera-code-scanner/issues/79#issuecomment-1276041982